plugin/src/main/java/com/playmonumenta/redissync/DataEventListener.java

```diff
@@ -523,9 +553,38 @@ public void playerDataLoadEvent(PlayerDataLoadEvent event) {
 			event.setData(nbtTagCompound);
 
 			mLogger.fine(() -> "Processing PlayerDataLoadEvent took " + (System.currentTimeMillis() - startTime) + " milliseconds on main thread");
-		} catch (IOException | InterruptedException | ExecutionException ex) {
-			mLogger.severe("Failed to load player data: " + ex);
-			ex.printStackTrace();
+		} catch (Throwable ex) {
+			mLogger.severe("!!! Failed to load player data !!!");
+			mLogger.log(Level.SEVERE, "Exception", ex);
+
+			final var rootPath = MonumentaRedisSync.getInstance().getDataFolder().toPath()
+				.resolve("data-fail-report-%s-%s-%s".formatted(
+					new SimpleDateFormat("yyyy-MM-dd-HH-mm-ss-SSS").format(Instant.now()),
+					player.getName(),
+					player.getUniqueId()
+				));
+
+			mLogger.severe("Writing data files to for analysis..." + rootPath);
+
+			try {
+				Files.createDirectories(rootPath);
+				trySave(rootPath, "error.txt", dest -> Files.writeString(dest, exceptionToString(ex)));
+				trySave(rootPath, "data.nbt", dest -> Files.write(dest, dataFuture.get()));
+				trySave(rootPath, "plugin_data.nbt", dest -> Files.writeString(dest, pluginDataFuture.get()));
+				trySave(rootPath, "score.nbt", dest -> Files.writeString(dest, scoreFuture.get()));
+				trySave(rootPath, "shard", dest -> {
+					Files.createDirectories(dest);
+					for (final var ent : shardDataFuture.get().entrySet()) {
+						Files.writeString(dest.resolve(ent.getKey() + ".json"), ent.getValue());
+					}
+				});
+			} catch (IOException e) {
```

Comment:
Log e here and I think we're good (just in case it tells us something like "hey, you forgot to make that directory writable, go fix it server ops")